fe/fe-core/src/main/java/com/starrocks/sql/plan/PlanFragmentBuilder.java

```diff
@@ -758,11 +758,29 @@ public PlanFragment visitPhysicalOlapScan(OptExpression optExpr, ExecPlan contex
             }
 
             // set slot
+            Map<String, Column> mvColumnMap = Maps.newHashMap();
+            boolean scanBaseTable = node.getSelectedIndexId() == referenceTable.getBaseIndexId();
+            if (!scanBaseTable) {
+                MaterializedIndexMeta indexMeta = referenceTable.getIndexMetaByIndexId(node.getSelectedIndexId());
+                mvColumnMap = indexMeta.getSchema().stream()
+                        .filter(x -> CollectionUtils.isNotEmpty(x.getRefColumns()))
+                        .collect(Collectors.toMap(x -> x.getRefColumns().get(0).getColumnName().toLowerCase(), x -> x));
+            }
             for (Map.Entry<ColumnRefOperator, Column> entry : node.getColRefToColumnMetaMap().entrySet()) {
                 SlotDescriptor slotDescriptor =
                         context.getDescTbl().addSlotDescriptor(tupleDescriptor, new SlotId(entry.getKey().getId()));
-                slotDescriptor.setColumn(entry.getValue());
-                slotDescriptor.setIsNullable(entry.getValue().isAllowNull());
+                Column column = entry.getValue();
+                if (scanBaseTable || !mvColumnMap.containsKey(column.getName())) {
+                    slotDescriptor.setColumn(entry.getValue());
+                    slotDescriptor.setIsNullable(entry.getValue().isAllowNull());
+                } else {
+                    // Replace the column to mv column for special case:
+                    // If the MV is for aggregate table, and the query doesn't contain aggregate function
+                    // The MV would be chosen but the columns would not be rewritten
+                    Column mvColumn = mvColumnMap.get(column.getName());
+                    slotDescriptor.setColumn(mvColumn);
+                    slotDescriptor.setIsNullable(mvColumn.isAllowNull());
```

Comment:
It seems weird to place this here. Is it possible that the root cause lies within the MaterializedViewRule?

Reply:
yes, but it's pretty complicated to fix it in the rule.

The root cause is, previously in statement `CREATE MATERIALIZED VIEW v0 AS SELECT t1_17.c_1_0, t1_17.c_1_1, SUM(t1_17.c_1_2)`, the `SUM(t1_17.c_1_2)` must has an alias, but right now it doesn't, and assigned an automatic name for it.